pkg/controller/common/certificates/reconcile.go

```diff
@@ -32,7 +33,8 @@ type Reconciler struct {
 	K8sClient      k8s.Client
 	DynamicWatches watches.DynamicWatches
 
-	Object        metav1.Object                     // owner for the TLS certificates (eg. Elasticsearch, Kibana)
+	Owner runtime.Object // owner for the TLS certificates (eg. Elasticsearch, Kibana)
```

Comment:
Indeed, sorry I should have made it clear in a PR comment.
The reason is that we need to access the `Kind`, which is not available in a `metav1.Object`.

I feel like a comment in the Github PR would have been useful, but a comment in the code would maybe be strange and not bring much value (`// we need a runtime.Object and not a metav1.Object here because we use the Kind deep down`).

Reply:
Absolutely, a comment would feel out of place.